SoftLayer/CLI/hardware/detail.py

```diff
@@ -107,6 +110,24 @@ def cli(env, identifier, passwords, price):
             pass_table.add_row([item['username'], item['password']])
         table.add_row(['remote users', pass_table])
 
+    if components:
+        components = hardware.get_components(identifier)
+        components_table = formatting.Table(['name', 'version', 'date', 'component-type'])
```

Comment:
```suggestion
                components_table = formatting.Table(['name', 'Firmware version', 'Firmware build date', 'Type'])
```
maybe adding the id too